docs/proposals/prop-apb-gen-creds.md

```diff
@@ -0,0 +1,41 @@
+# APB Generate Credentials Secret Proposal
+
+## Introduction
+
+This proposal aims to improve the broker process by which we extract
+credentials from bindable [Ansible Playbook
+Bundles (APB)](https://github.com/ansibleplaybookbundle/ansible-playbook-bundle)
+by allowing the APB to generate a secret with needed credentials instead of
+using `kubectl exec` into the running pod to grab credentials from a file.
+
+Reference Issue #544
+
+## Problem Description
+
+The broker should seek to use available kubernetes API calls to get necessary
+information whenever possible instead of relying on `kubectl exec`. Now that
+the broker supplies pod name and namespace information to running APBs via the
+[Kubernetes Downward
+API](https://kubernetes.io/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/),
+APBs have all they need to generate secrets inside the APB sandbox namespace to
+be retrieved later by the broker.
```

Comment:
I would split the last sentence fragment so that it is more clear:

APBs will have all they need to generate secrets inside the APB sandbox namespace. The secret will be retrieved by the broker after the pod finishes.